src/Generator/Passes/DelegatesPass.cs

```diff
@@ -35,33 +43,135 @@ public DelegatesPass()
             VisitOptions.VisitTemplateArguments = false;
         }
 
+        private struct Typedef
```

Comment:
I can see the name-space of the stored typedef is the same as this context here. So having the typedef only is enough, remove this struct.